### ISSUE TITLE
Null-check operator on RangeSlider's _startSemanticsNode

### ISSUE THREAD

**Issue by r-i-c-o** [flutter/flutter#141953]
### Steps to reproduce

Try to run the sample code. We could not reproduce it on a local machine.

### Expected results

No problems with the semantics part of a `RangeSlider`

### Actual results

The issue appears in Sentry in release mode. The `_startSemanticsNode` field somehow became `null`. This seems to be related with https://github.com/flutter/flutter/issues/91029

### Code sample

<details open><summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        colorScheme: ColorScheme.fromSeed(seedColor: Colors.deepPurple),
        useMaterial3: true,
      ),
      home: const MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatelessWidget {
  const MyHomePage({super.key, required this.title});

  final String title;

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Center(
        child: ElevatedButton(
          onPressed: () {
            showModalBottomSheet(
              context: context,
              builder: (context) {
                return _BottomSheetRangeSlider();
              },
            );
          },
          child: Text("Show range slider"),
        ),
      ),
    );
  }
}

class _BottomSheetRangeSlider extends StatefulWidget {
  const _BottomSheetRangeSlider({super.key});

  @override
  State<_BottomSheetRangeSlider> createState() => _BottomSheetRangeSliderState();
}

class _BottomSheetRangeSliderState extends State<_BottomSheetRangeSlider> {
  double _s = 1;
  double _e = 100;
  @override
  Widget build(BuildContext context) {
    return SingleChildScrollView(
      child: Column(
        children: [
          Column(
            children: [
              Padding(
                padding: const EdgeInsets.all(8.0),
                child: RangeSlider(
                  values: RangeValues(_s, _e),
                  min: 1,
                  max: 100,
                  divisions: 50,
                  onChanged: (RangeValues values) {
                    setState(() {
                      _s = values.start;
                      _e = values.end;
                    });
                  },
                ),
              ),
            ],
          ),
        ],
      ),
    );
  }
}
```

</details>

### Logs

<details open><summary>Logs</summary>

```console
_TypeError: Null check operator used on a null value
  File "range_slider.dart", line 1700, in _RenderRangeSlider.assembleSemanticsNode
  File "object.dart", line 4947, in _SwitchableSemanticsFragment.compileChildren
  File "object.dart", line 4871, in _SwitchableSemanticsFragment.compileChildren
  File "object.dart", line 4936, in _SwitchableSemanticsFragment.compileChildren
  File "object.dart", line 3601, in RenderObject._updateSemantics
  File "object.dart", line 1277, in PipelineOwner.flushSemantics
  File "object.dart", line 1282, in PipelineOwner.flushSemantics
  File "binding.dart", line 598, in RendererBinding.drawFrame
  File "binding.dart", line 986, in WidgetsBinding.drawFrame
  File "binding.dart", line 457, in RendererBinding._handlePersistentFrameCallback
  File "binding.dart", line 1325, in SchedulerBinding._invokeFrameCallback
  File "binding.dart", line 1255, in SchedulerBinding.handleDrawFrame
  File "binding.dart", line 1113, in SchedulerBinding._handleDrawFrame
  File "hooks.dart", line 312, in _invoke
  File "platform_dispatcher.dart", line 383, in PlatformDispatcher._drawFrame
  File "hooks.dart", line 283, in _drawFrame
```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
[√] Flutter (Channel stable, 3.16.7, on Microsoft Windows [Version 10.0.19045.3930], locale ru-RU)
[√] Windows Version (Installed version of Windows is version 10 or higher)
[√] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
[X] Chrome - develop for the web (Cannot find Chrome executable at .\Google\Chrome\Application\chrome.exe)
    ! Cannot find Chrome. Try setting CHROME_EXECUTABLE to a Chrome executable.
[√] Visual Studio - develop Windows apps (Visual Studio Community 2022 17.4.4)
[√] Android Studio (version 2023.1)
[√] IntelliJ IDEA Community Edition (version 2022.3)
[√] Connected device (3 available)
[√] Network resources

```

</details>

**Comment by TahaTesser** [comment]
Hi @r-i-c-o 
Thanks for the report.  I can reproduce this locally.

### Steps to reproduce 
1. Run  `RangeSlider` code sample
2. Put the app in the background and enable `Voice Over`, switch back to Flutter app.
3. As you soon as you switch to the Flutter app. you get this null check error

### Code sample

```dart
import 'package:flutter/material.dart';

void main() => runApp(const MyApp());

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      debugShowCheckedModeBanner: false,
      home: Material(
        child: RangeSlider(
          values: const RangeValues(40, 80),
          max: 100,
          onChanged: (RangeValues newValue) {},
        ),
      ),
    );
  }
}

```

<details>
<summary>expand to view the logs</summary> 

```console
Launching lib/main.dart on Taha’s iPhone in debug mode...
Automatically signing iOS for device deployment using specified development team in Xcode project: X8NNQ9CYL2
Xcode build done.                                           11.3s
You may be prompted to give access to control Xcode. Flutter uses Xcode to run your app. If access is not allowed, you can change this through your Settings > Privacy & Security > Automation.
The Dart VM Service was not discovered after 60 seconds. This is taking much longer than expected...
Open the Xcode window the project is opened in to ensure the app is running. If the app is not running, try selecting "Product > Run" to fix the problem.

[ERROR:flutter/shell/platform/darwin/graphics/FlutterDarwinContextMetalImpeller.mm(42)] Using the Impeller rendering backend.
Connecting to VM Service at ws://127.0.0.1:55214/s9nBWrIkYhE=/ws

════════ Exception caught by scheduler library ═════════════════════════════════
The following _TypeError was thrown during a scheduler callback:
Null check operator used on a null value

When the exception was thrown, this was the stack:
#0      _RenderRangeSlider.assembleSemanticsNode (package:flutter/src/material/range_slider.dart:1703:28)
#1      _SwitchableSemanticsFragment.compileChildren (package:flutter/src/rendering/object.dart:4962:13)
#2      _SwitchableSemanticsFragment.compileChildren (package:flutter/src/rendering/object.dart:4951:16)
#3      _SwitchableSemanticsFragment.compileChildren (package:flutter/src/rendering/object.dart:4951:16)
#4      _SwitchableSemanticsFragment.compileChildren (package:flutter/src/rendering/object.dart:4888:18)
#5      _SwitchableSemanticsFragment.compileChildren (package:flutter/src/rendering/object.dart:4888:18)
#6      _SwitchableSemanticsFragment.compileChildren (package:flutter/src/rendering/object.dart:4888:18)
#7      _SwitchableSemanticsFragment.compileChildren (package:flutter/src/rendering/object.dart:4888:18)
#8      _SwitchableSemanticsFragment.compileChildren (package:flutter/src/rendering/object.dart:4888:18)
#9      _SwitchableSemanticsFragment.compileChildren (package:flutter/src/rendering/object.dart:4951:16)
#10     _RootSemanticsFragment.compileChildren (package:flutter/src/rendering/object.dart:4666:16)
#11     RenderObject._updateSemantics (package:flutter/src/rendering/object.dart:3630:25)
#12     PipelineOwner.flushSemantics (package:flutter/src/rendering/object.dart:1303:16)
#13     PipelineOwner.flushSemantics (package:flutter/src/rendering/object.dart:1308:15)
#14     RendererBinding.drawFrame (package:flutter/src/rendering/binding.dart:584:25)
#15     WidgetsBinding.drawFrame (package:flutter/src/widgets/binding.dart:991:13)
#16     RendererBinding._handlePersistentFrameCallback (package:flutter/src/rendering/binding.dart:443:5)
#17     SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:1386:15)
#18     SchedulerBinding.handleDrawFrame (package:flutter/src/scheduler/binding.dart:1311:9)
#19     SchedulerBinding._handleDrawFrame (package:flutter/src/scheduler/binding.dart:1169:5)
#20     _invoke (dart:ui/hooks.dart:312:13)
#21     PlatformDispatcher._drawFrame (dart:ui/platform_dispatcher.dart:399:5)
#22     _drawFrame (dart:ui/hooks.dart:283:31)


```

</details>

**Comment by TahaTesser** [comment]
> If this problem is related to the `Semantics` part (`assembleSemanticsNode` method specifically), can it be fixed by wrapping `RangeSlider` in `ExcludeSemantics` as a temporary solution?

This will disable accessibility features such as talk back, voice over.
